vowpalwabbit/learner.h

```diff
@@ -209,18 +257,46 @@ typedef void (*tend_example)(vw& all, void* d, example& ec);
   //called after learn example for each example.  Explicitly not recursive.
   inline void finish_example(vw& all, example& ec)
   { finish_example_fd.finish_example_f(all, finish_example_fd.data, ec);}
+  //called after learn example for each example.  Explicitly not recursive.
+  inline void finish_example(vw& all, multi_ex& ec)
+  {
+    finish_example_fd.finish_multi_ex_f(all, finish_example_fd.data, ec);
+  }
   void set_finish_example(void (*f)(vw& all, T&, example&))
   { finish_example_fd.data = learn_fd.data;
     finish_example_fd.finish_example_f = (tend_example)f;
   }
+  void set_finish_example(void(*f)(vw& all, T&, multi_ex&))
+  {
+    finish_example_fd.data = learn_fd.data;
+    finish_example_fd.finish_multi_ex_f = (tend_multi_ex)f;
+  }
+  //called to check if an example is a test example.  Used in creating multiline example
+  inline bool is_test_example(example& ec)
+  {
+    return test_example_fd.test_example_f(ec);
+  }
+  void set_test_example(bool(*f)(example&))
+  {
+    test_example_fd.test_example_f = (ttest_example)f;
+  }
 
-  friend learner<T>& init_learner<>(T*, base_learner*, void (*l)(T&, base_learner&, example&),
-                                      void (*pred)(T&, base_learner&, example&), size_t, prediction_type::prediction_type_t);
+  friend learner<T>& init_learner<>(T*, base_learner*, 
```

Comment:
Why should we need a learner that supports both single and multline examples?  It seems like this is cuttable?